content/user/apps/Session-metadata.md

```diff
@@ -0,0 +1,68 @@
+
+### Examples
+ 
+If you use system properties for [Java](installation-javaconfig.html), include an additional entry in the line where you add your `javaagent` flag. In this case, you will set the property `contrast.application.session_metadata` to a set of key-value pairs that identify your test run. 
+
+```
+-Dcontrast.application.session_metadata="branchName=feature/some-new-thing,committer=Jane,repository=Contrast-Java"
+```
+ 
+As a [.NET](installation-netconfig.html) developer using *app.config* or *web.config*, you can add an entry to your configuration to specify this property. 
+
+```
+<?xml version="1.0"?>
+<configuration>
+  <connectionStrings />
+  <appSettings>
+    <add key="contrast.application.session_metadata" value="branchName=feature/some-new-thing,committer=Jane,repository=Contrast-DotNet" />
+```
+ 
+If you use YAML configuration, like most [Node](installation-nodeconfig.html) and [Ruby](installation-rubyconfig.html) agents, you can add an additional entry to your *contrast_security.yaml* file. 
+
+```
+application:
+ session_metadata:branchName=feature/some-new-thing,committer=Jane,repository=Contrast-Ruby
+```
+ 
+### Continuous integration 
+
+If you don't want to set these fields locally, you can also edit your continuous integration (CI) build scripts to set these for you. In Jenkins, you can pull the build number using the `BUILD_NUMBER` property. 
+
+```
+-Dcontrast.application.session_metadata="branchName=feature/some-new-thing,committer=Jane,repository=Contrast-Java,buildNumber=$BUILD_NUMBER"
+```
+
+If you use plugins in your CI, like the [Contrast Jenkins Plugin](tools-ci.html#jenkins), you can use these to make even more dynamic updates. 
+
+```
+-Dcontrast.application.session_metadata="branchName=$GIT_BRANCH,committer=$GIT_COMMITTER_NAME,commitHash=$GIT_COMMIT_HASH,repository=$GIT_URL,buildNumber=$BUILD_NUMBER"
+```
+
+## View Data by Application
+
+To see the session metadata reported by the agent, go to your application's **Vulnerabilities** tab. The data for each vulnerability is displayed in the grid and the timeline. 
+
+Use the **View By** menu above the timeline to filter the data by the properties that you included in your agent configuration. This updates the values shown in the **Seen By** column in the grid. Use the filter for the grid column to refine the results. 
+
+<a href="assets/images/Seen-by-filter.png" rel="lightbox" title="Filter vulnerabilities by categories of session metadata"><img class="thumbnail" src="assets/images/Seen-by-filter.png"/></a>
+
+> **Note:** To see vulnerabilities that aren't associated with any session metadata, select **Disassociated** in the **View By** menu. 
+
+Your selections also update the data shown in the timeline, which you can view by **Severity** or **Discovery**. Hover over the trend lines in the grid for a breakdown of the data at that point in time. 
```

Comment:
The meaning of "viewing by severity" is readily apparent. I'm not so sure "viewing by discovery" is. Some options could be to add a subsequent short description, like "…, which shows when vulnerabilities were found." or if you'd rather work the name into the description like "view by vulnerability *Severity* or by the *Discovery* date of a vulnerability" or obviously some third option, up to you.

Reply:
Got it. 

I've bolded the key words in the following paragraph where it explains what data is included in each view. Does that give enough clarity, or do we need more in this paragraph as well?